pkg/controller/argocd/role.go

```diff
@@ -55,19 +57,19 @@ func newClusterRole(name string, rules []v1.PolicyRule, cr *argoprojv1a1.ArgoCD)
 
 // reconcileRoles will ensure that all ArgoCD Service Accounts are configured.
 func (r *ReconcileArgoCD) reconcileRoles(cr *argoprojv1a1.ArgoCD) (role *v1.Role, err error) {
```

Comment:
This function probably should not return `role`.

Reply:
These roles are needed when we reconcile `RoleBindings`.

Reply:
I am confused about this method as well. It basically never returns any role, if I interpret the code correctly.

if `r.reconcileRole()` returns `err != nil`, `role` would be `nil` as well, right? So basically, all it ever returns is `nil` for `role` return value. :thinking:

Reply:
We actually don't need the list of `roles` when we are calling the `reconcileRole` explicitly and just need to make sure that the role was created without any errors!! 
But when we call `reconcileRole` from within the `reconcileRoleBindings`, we need the list of roles that were created which would eventually be used to create the `RoleBindings`!

This would make more sense when we refactor the ArgoCD Operator code, which wouldn't reconcile the `Role/RoleBinding/ServiceAccount` individually, rather, it would call a function something like `reconcileArgoCDPermissions` and within the function, it would reconcile all the resources in the order `ServiceAccount -> Roles -> RoleBindings`